### ISSUE TITLE
WMCore campaign SiteWhitelist should not inherit RSEs defined under the unified secondaries section

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#10332]
**Impact of the new feature**
MSTransferor and Unified code 

**Is your feature request related to a problem? Please describe.**
Issue has been reported and discussed in this JIRA ticket: https://its.cern.ch/jira/browse/CMSCOMPPR-18418

The whole pileup management - considering the campaigns - need some real rework. This issue is only meant to have a short term fix such that some workflows can properly go through in the system.

**Describe the solution you'd like**
When creating the WMCore campaign document - which gets uploaded to central CouchDB - we should not use the pileup location (provided under the `secondaries` key) as a set of top level `SiteWhitelist`. In other words, expected pileup data location should not affect the campaign SiteWhitelist, which is where all workflows under that campaign are eligible to run.

In short, these lines should be removed: https://github.com/dmwm/WMCore/blob/master/bin/adhoc-scripts/parseUnifiedCampaigns.py#L150-L153

Note that the same changes must be provided to the P&R repository as well, likely on this module:
https://github.com/CMSCompOps/WmAgentScripts/blob/master/campaignAPI.py

Just an important note:
* SiteWhitelist: is a list of PSNs (CPU resources)
* SecondaryLocation or data location: is a list of RSEs or PNNs (storage resources) 

**Describe alternatives you've considered**
none

**Additional context**
Refer to the first description in: https://github.com/dmwm/WMCore/issues/9139 for a documentation on the WMCore campaign schema.

This issue became worse with the changes provided in this PR (already in production):
https://github.com/dmwm/WMCore/pull/10243

**Comment by amaltaro** [comment]
@haozturk Hasan, can you please review this issue and complement anything that I might be missing from the discussion we had this afternoon? Thanks

**Comment by haozturk** [comment]
Thanks @amaltaro  your description looks very accurate to me. I did the necessary changes on the Unified side: https://github.com/CMSCompOps/WmAgentScripts/pull/801/files
